crates/sources/src/loader.rs

```diff
@@ -764,24 +836,58 @@ impl<F: Fetcher> Loader<F> {
         );
     }
 
-    fn load_capture_endpoint<'s>(
+    async fn load_capture_endpoint<'s>(
         &'s self,
-        _scope: Scope<'s>,
-        endpoint: models::CaptureEndpoint,
+        scope: Scope<'s>,
+        mut endpoint: models::CaptureEndpoint,
     ) -> Option<serde_json::Value> {
         use models::CaptureEndpoint::*;
+
+        // Map a URL reference into its absolute form, and load it.
+        if let AirbyteSource(models::ConnectorConfig {
+            image,
+            config: models::Config::Url(relative),
+        }) = endpoint
+        {
+            let absolute = self.fallible(scope, scope.resource().join(&relative))?;
+            self.load_import(scope, &absolute, models::ContentType::Config)
+                .await;
+
+            endpoint = AirbyteSource(models::ConnectorConfig {
+                image,
+                config: models::Config::Url(models::RelativeUrl::new(absolute.to_string())),
```

Comment:
(Most likely due to my lack of knowledge in the catalog building process and Rust) 
It looks to me that since the process is `await` on the `load_import`, the config file should be loaded when the execution reaches this line, so is it possible to perform the `to_inline` here, instead of storing the URL... 
I feel there should be some reasons of not doing it, but not sure about the exact.

Reply:
The trouble comes if there are _multiple_ references to the config which are reached concurrently. The first call to `load_import` will indeed block and you could immediately inline it's result. But, the _second_ call observes that a fetch is already happening and will return immediately (not waiting for the fetch to complete).

It would be possible to rework this further and have the second call enqueue a one-shot channel on which it awaits, where the first call notifies all such channels on completion. This is a deeper change, though, which didn't seem worth it to me.

Reply:
Got it. Thanks for the explanations.